runtime/src/main/java/org/corfudb/protocols/CorfuProtocolServerErrors.java

```diff
@@ -0,0 +1,151 @@
+                .build();
+    }
+
+    /**
+     * Returns the Protobuf representation of a OVERWRITE error.
+     * @param causeId   The ID of the OverwriteCause in OverwriteException.
+     */
+    public static ServerErrorMsg getOverwriteErrorMsg(int causeId) {
+        return ServerErrorMsg.newBuilder()
+                .setOverwriteError(OverwriteErrorMsg.newBuilder()
+                        .setOverwriteCauseId(causeId)
+                        .build())
+                .build();
+    }
+
+    /**
+     * Returns the Protobuf representation of a DATA_OUTRANKED error.
+     */
+    public static ServerErrorMsg getDataOutrankedErrorMsg() {
+        return ServerErrorMsg.newBuilder()
+                .setDataOutrankedError(DataOutrankedErrorMsg.getDefaultInstance())
+                .build();
+    }
+
+    // Method below will be added in later PR with other LogUnit updates
+    // public static ServerErrorMsg getValueAdoptedErrorMsg(ReadResponse rr)
+
+    /**
+     * Returns the Protobuf representation of a DATA_CORRUPTION error.
+     * @param address   The address that caused the error.
+     */
+    public static ServerErrorMsg getDataCorruptionErrorMsg(long address) {
+        return ServerErrorMsg.newBuilder()
+                .setDataCorruptionError(DataCorruptionErrorMsg.newBuilder()
+                        .setAddress(address)
+                        .build())
+                .build();
+    }
+
+    /**
+     * Returns the Protobuf representation of a BOOTSTRAPPED error.
+     */
+    public static ServerErrorMsg getBootstrappedErrorMsg() {
+        return ServerErrorMsg.newBuilder()
+                .setBootstrappedError(BootstrappedErrorMsg.getDefaultInstance())
+                .build();
+    }
+
+    /**
+     * Returns the Protobuf representation of a NOT_BOOTSTRAPPED error.
+     */
+    public static ServerErrorMsg getNotBootstrappedErrorMsg() {
+        return ServerErrorMsg.newBuilder()
+                .setNotBootstrappedError(NotBootstrappedErrorMsg.getDefaultInstance())
+                .build();
+    }
+
+    /**
+     * Returns the Protobuf representation of a SERVER_EXCEPTION error.
+     * This is used for errors that do not fit with the above types.
+     * @param throwable   The underlying throwable cause.
+     */
+    public static ServerErrorMsg getUnknownErrorMsg(Throwable throwable) {
+        UnknownErrorMsg.Builder unknownErrorBuilder = UnknownErrorMsg.newBuilder();
+
+        try (ByteString.Output bso = ByteString.newOutput()) {
+            try (ObjectOutputStream oos = new ObjectOutputStream(bso)) {
+                oos.writeObject(throwable);
+                unknownErrorBuilder.setThrowable(bso.toByteString());
+            }
+        } catch (Exception ex) {
+            //TODO(Zach): Any other handling? Wrap around SerializationFailedException?
+            log.error("getUnknownErrorMsg: error=[{}, {}] " +
+                    "while serializing throwable={}", ex, ex.getCause(), throwable);
```

Comment:
the last parameter of log.error can just be the throwable. This usage is more for log.warn()

Reply:
Done.